[Feat] Add 4o-mini to model list (#627)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md

    - gpt-3.5-turbo-1106
    - gpt-3.5-turbo-0125
    - gpt-4o-2024-05-13
    - gpt-4o-mini-2024-07-18

>     - gpt-4o-mini

Reviewer: Can we also double check the other models, if they are still up-to-date?

Author: Just double checked, everything is up to date so far 👍

### Chart generation

> For chart creation with OpenAI, the **gpt-4o-mini** and **gpt-3.5** model series have lower price points and faster speeds for providing answers, but do not offer sophisticated charting.

Reviewer 2: ```suggestion
For basic chart creation with OpenAI, the **gpt-4o-mini** and **gpt-3.5** model series have lower price points and faster speeds, but they do not offer sophisticated charting.
```

> Consider upgrading to the **gpt-4o** and **gpt-4** model series for more demanding tasks. While they are part of a more capable GPT model series, they come at a higher cost.

Reviewer 2: ```suggestion
Consider upgrading to the **gpt-4o** and **gpt-4** model series for more demanding tasks. The downside of using these models is that they come at a higher cost.
```

=== "OpenAI"

    - gpt-3.5-turbo `default`: Ideal for smaller requests and simple specifications. However, due to the complexity of dashboard creation, `gpt-3.5-turbo` often produces incomplete responses for larger tasks.

>     - gpt-4o: Most ideal for dashboard generation tasks. Excels in dashboard creation tasks, providing stable and complete responses. Compared to gpt-4-turbo, gpt-4o is much faster and cheaper.

Reviewer 2: ```suggestion
    - gpt-4o: Most ideal for dashboard generation tasks, providing stable and complete responses. Compared to gpt-4-turbo, gpt-4o it is much faster and cheaper.
```

    - gpt-4-turbo: Excels in dashboard creation tasks, providing stable and complete responses.

>     - gpt-4o-mini: Ideal for smaller requests and simple specifications. Most cost-effective. However, due to the complexity of dashboard creation, its performance is not stable.

Reviewer 2: ```suggestion
    - gpt-4o-mini: Ideal for smaller requests and simple specifications. Most cost-effective. However, due to the complexity of dashboard creation, its performance is not guaranteed to be reliable.
```

Reviewer 2: I went with "reliable" rather than "stable" is that OK?
